res/skins/QMLDemo/EffectSlot.qml

```diff
@@ -0,0 +1,196 @@
+import "." as Skin
+import Mixxx 0.1 as Mixxx
+import QtQuick 2.12
+import "Theme"
+
+Item {
+    id: root
+
+    property var slot: Mixxx.EffectsManager.getEffectSlot(1, unitNumber, effectNumber)
+    property int unitNumber // required
+    property int effectNumber // required
+    property bool expanded: false
+    readonly property string group: slot.group
+    property real maxSelectorWidth: 300
+
+    height: 50
+
+    Item {
+        id: selector
+
+        anchors.left: parent.left
+        anchors.top: parent.top
+        anchors.bottom: parent.bottom
+        width: Math.min(root.width, root.maxSelectorWidth)
+
+        Skin.ControlButton {
+            id: effectEnableButton
+
+            anchors.left: parent.left
+            anchors.top: parent.top
+            anchors.bottom: parent.bottom
+            anchors.margins: 5
+            width: 40
+            group: root.group
+            key: "enabled"
+            toggleable: true
+            text: "ON"
+            activeColor: Theme.effectColor
+        }
+
+        Skin.ComboBox {
+            id: effectSelector
+
+            anchors.top: parent.top
+            anchors.bottom: parent.bottom
+            anchors.left: effectEnableButton.right
+            anchors.right: effectMetaKnob.left
+            anchors.margins: 5
+            // TODO: Add a way to retrieve effect names here
+            textRole: "display"
+            model: Mixxx.EffectsManager.visibleEffectsModel
+            onActivated: {
+                const effectId = model.get(index).effectId;
+                if (root.slot.effectId != effectId)
+                    root.slot.effectId = effectId;
+
+            }
+            Component.onCompleted: root.slot.onEffectIdChanged()
+
```

Comment:
This makes sure the combobox shows the effect name after starting mixxx (Mixxx remembers effects persistently). Otherwise the effect slot look empty even though it's not.